Tests/ApolloInternalTestHelpers/MockURLProtocol.swift

```diff
@@ -0,0 +1,57 @@
+import Foundation
+
+public class MockURLProtocol<RequestProvider: MockRequestProvider>: URLProtocol {
+  
+  override class public func canInit(with request: URLRequest) -> Bool {
+    return true
+  }
+  
+  override class public func canonicalRequest(for request: URLRequest) -> URLRequest {
+    return request
+  }
+  
+  override public func startLoading() {
+    guard let url = request.url,
+          let handler = RequestProvider.requestHandlers[url] else {
+      fatalError("No URL available for URLRequest.")
+    }
+    
+    defer {
+      RequestProvider.requestHandlers.removeValue(forKey: url)
+    }
+    
+    // Provide random response delay between 0 and 1 seconds
+    Thread.sleep(forTimeInterval: Double.random(in: 0.0...1.0))
```

Comment:
need to re-evaluate handling the response delay since this seems to be causing some issues when run on the CI server.